src/AutoRest.CSharp/MgmtTest/Generation/ScenarioTestWriter.cs

```diff
@@ -0,0 +1,347 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using AutoRest.CSharp.Generation.Writers;
+using AutoRest.CSharp.Mgmt.AutoRest;
+using AutoRest.CSharp.Input;
+using AutoRest.CSharp.Mgmt.Output;
+using AutoRest.CSharp.Mgmt.Models;
+using AutoRest.CSharp.Utilities;
+using AutoRest.CSharp.Common.Generation.Writers;
+using Azure.ResourceManager.Resources;
+
+namespace AutoRest.CSharp.MgmtTest.Generation
+{
+    /// <summary>
+    /// Code writer for resource test.
+    /// </summary>
+    internal class ScenarioTestWriter: ClientWriter
+    {
+        protected TestDefinitionModel testDef;
+        protected CodeWriter writer;
```

Comment:
```suggestion
        protected CodeWriter _writer;
```
Also do we need these two fields protected? Is there any class inheriting this class?

Reply:
currently no inheriting class. changing to private